src/trio/_core/_run.py

```diff
@@ -734,13 +750,48 @@ def deadline(self) -> float:
         this can be overridden by the ``deadline=`` argument to
         the :class:`~trio.CancelScope` constructor.
         """
+        if self._is_relative is True:
+            raise RuntimeError(
+                "unentered relative cancel scope does not have an absolute deadline",
+            )
         return self._deadline
 
     @deadline.setter
     def deadline(self, new_deadline: float) -> None:
+        if self._is_relative is True:
+            raise RuntimeError(
+                "unentered relative cancel scope does not have an absolute deadline",
+            )
```

Comment:
Here too, I'd emit a deprecation warning but set the deadline as expected, and if-not-entered also set ~~is_relative=False~~ _relative_deadline=inf.

Reply:
For this one I have no problems with raising a warning instead.